### PR TITLE
VPA: Add tests for selfRegistration() and filterVPAs()

### PR DESCRIPTION
**Pull request by adrianmoisey**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

I'm working on another PR
(https://github.com/kubernetes/autoscaler/pull/6788) and realised that those changes will be touching code that is untested.

I decided to go back and add some tests before continuing work in that PR.

#### Which issue(s) this PR fixes:

N/A

#### Special notes for your reviewer:

I'm unsure if these tests match the current style of tests in the VPA. I'm happy to change if needed.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder_test.go [hunk 1]
```diff
@@ -523,3 +523,52 @@ func TestClusterStateFeeder_InitFromHistoryProvider(t *testing.T) {
...
+	expectedResult := []*vpa_types.VerticalPodAutoscaler{vpa2}
+
+	// Run the filterVPAs function
+	result := filterVPAs(feeder, allVpaCRDs)
+
+	if len(result) != len(expectedResult) {
+		t.Fatalf("expected %d VPAs, got %d", len(expectedResult), len(result))
+	}
+
+	for i, vpa := range result {
```
[the remark this answers is not in the document]
Comment (adrianmoisey) [reply vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder_test.go]: Great suggestion, thanks!

I've done that in 9c0941f103

**Comment by adrianmoisey** [comment]
Hey @raywainman 
Any chance you have some time to look at this again?

**Comment by adrianmoisey** [comment]
Hey @raywainman 
It looks like the `/lgtm` didn't work. I wonder if the "thanks!" tripped the bot up.
Can you try again?

**Comment by voelzmo** [comment]
/lgtm
/approve
Thanks!

**Pull request merged**